subgraphs/kwenta/src/mappings/handlers.ts

```diff
@@ -0,0 +1,491 @@
+  const marketKey = event.params.marketKey.toString();
+  const sdk = SDK.initializeFromEvent(
+    conf,
+    new Pricer(),
+    new TokenInit(),
+    event
+  );
+
+  // check that it's a v1 market before adding
+  if (marketKey.endsWith("PERP")) {
+    log.info("New V2 market added: {}", [marketKey]);
+
+    const pool = sdk.Pools.loadPool(event.params.market);
+    if (!pool.isInitialized) {
+      const token = sdk.Tokens.getOrCreateToken(
+        NetworkConfigs.getSUSDAddress()
+      );
+      pool.initialize(marketKey, marketKey, [token], null, "chainlink");
+
+      // keeper dynamic fees
+      pool.setPoolFee(LiquidityPoolFeeType.DYNAMIC_LP_FEE, null);
+    }
+
+    // perps v2 market
+    PerpsV2Market.create(event.params.market);
+  }
+}
+
+/*
+  This function is fired when a new smart margin account is created.
+  We are storing the new smart margin account with it's owner address for future reference.
+*/
+export function handleNewAccountSmartMargin(
+  event: NewSmartMarginAccountEvent
+): void {
+  // create a new entity to store the cross-margin account owner
+  const smAccountAddress = event.params.account as Address;
+  let smartMarginAccount = _SmartMarginAccount.load(smAccountAddress.toHex());
+  const sdk = SDK.initializeFromEvent(
+    conf,
+    new Pricer(),
+    new TokenInit(),
+    event
+  );
+  if (smartMarginAccount == null) {
+    smartMarginAccount = new _SmartMarginAccount(smAccountAddress.toHex());
+
+    const loadAccountResponse = sdk.Accounts.loadAccount(event.params.creator);
+    if (loadAccountResponse.isNewUser) {
+      const protocol = sdk.Protocol;
+      protocol.addUser();
+    }
+    smartMarginAccount.owner = loadAccountResponse.account.getBytesId();
+
+    smartMarginAccount.version = event.params.version;
+    smartMarginAccount.save();
+  }
+}
+
+/*
+ This function is fired when a Margin is transferred from or to a market position of an account.
+ If marginDelta > 0 then it is "deposit", else it is "withdraw".
+*/
+export function handleMarginTransferred(event: MarginTransferredEvent): void {
+  const marketAddress = dataSource.address();
+  const marginDelta = event.params.marginDelta;
+  const caller = event.params.account;
+
+  const sdk = SDK.initializeFromEvent(
+    conf,
+    new Pricer(),
+    new TokenInit(),
+    event
+  );
+  const pool = sdk.Pools.loadPool(marketAddress);
+  const token = sdk.Tokens.getOrCreateToken(NetworkConfigs.getSUSDAddress());
+
+  const loadAccountResponse = sdk.Accounts.loadAccount(caller);
+  const account = loadAccountResponse.account;
+  if (loadAccountResponse.isNewUser) {
+    const protocol = sdk.Protocol;
+    protocol.addUser();
+    pool.addUser();
+  }
+  const amounts = createTokenAmountArray(pool, [token], [marginDelta.abs()]);
+  // Deposit
+  if (marginDelta.gt(BIGINT_ZERO)) {
+    account.deposit(pool, amounts, BIGINT_ZERO);
+    pool.addInflowVolumeByToken(token, marginDelta.abs());
```

Comment:
In the schema, Inflow, Outflow and ClosedInflow Volumes are defined as trade volumes occured *at the time of position opened / closed*.
I believe `pool.addInflowVolumeByToken` (on position open); `pool.addOutflowVolumeByToken` and `pool.addClosedInflowVolumeByToken` (on position close) should be called inside `handlePositionModified`.